sui-execution/latest/sui-move-natives/src/tx_context.rs

```diff
@@ -54,3 +59,346 @@ pub fn derive_id(
         smallvec![Value::address(address)],
     ))
 }
+
```

Comment:
cc @tzakian for gas costing

Reply:
Yea, my take is `fresh_id` should match the cost of the existing one (`derive_id`), and now that you mention it I'm not sure why we just aren't using that native...

Reply:
`fresh_id` vs. `derive_id` :  `ids_created` will be tracked by the rust side, not Move code. `derive_id` was followed by an `ids_created += 1` in Move code (pretend Move had +=) so it is different. It did not make sense to reuse that and `fresh_id` with no arg seemed more reasonable. I could not change `derive_id` because it is also called by the test only `public fun last_created_object_id(self: &TxContext): address`. 
So don't know, it did not seem worth the effort trying to unify those... ideas welcome... maybe we should?